### PR TITLE
aqrdev: add 'restart' command

### PR DESCRIPTION
**Pull request by jecluis**
Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

**Comment by AvengerMoJo** [comment]
Not exactly related to this commit but I have two comment with aqrdev
1) the aqrdev require multiple pip install for the host may need to put into the setup.sh 
2) the doc also need to be updated accordingly.

**Comment by jecluis** [comment]
I don't know what you mean. All the dependencies for aqrdev (read, click) are on requirements.txt and are installed into the venv automatically, and the docs were updated accordingly to use aqrdev for what matters for the docs. What is it that is missing?

**Comment by AvengerMoJo** [comment]
> I don't know what you mean. All the dependencies for aqrdev (read, click) are on requirements.txt and are installed into the venv automatically, and the docs were updated accordingly to use aqrdev for what matters for the docs. What is it that is missing?

sorry about that I was testing just the deployment different machines and forget to run the setup-dev.sh nevermind ..

File: tools/aqrdev [hunk 1]
```diff
@@ -620,11 +630,16 @@ def _get_deployment_status(name: str) -> Dict[str, List[Tuple[str, str]]]:
 
     path: Path = _get_deployment_path(name)
     if not path.exists():
-        raise DeploymentPathNotFoundError()
+        raise DeploymentPathNotFoundError(
+            "deployment not found",
+            rc=errno.ENOENT
```
Comment (mgfritch) [inline tools/aqrdev:635]:
```suggestion
            rc=-errno.ENOENT
```

use negative errno int?

File: tools/aqrdev [hunk 2]
```diff
@@ -620,11 +630,16 @@ def _get_deployment_status(name: str) -> Dict[str, List[Tuple[str, str]]]:
...
+        )
 
     vagrantfile = path.joinpath("Vagrantfile")
     if not vagrantfile:
-        raise DeploymentNotFinishedError()
+        raise DeploymentNotFinishedError(
+            "deployment not finished", rc=errno.ENOENT
```
Comment (mgfritch) [inline tools/aqrdev:641]: I would almost prefer to reverse these args (e.g. errno appear first), but we'd need to extend the Error class todo that.

wdyt if I were to rework this via a PR #235 ?

**Review by mgfritch — APPROVED** [review]
couple nits, but otherwise lgtm!

File: tools/aqrdev (hunk 2 above)
Comment (jecluis) [reply tools/aqrdev:641]: > wdyt if I were to rework this via a PR #235 ?

sounds good to me!

**Comment by jhesketh** [comment]
So the documentation is a little light on, but should we be using ClickException for our base exception handler (https://click.palletsprojects.com/en/7.x/exceptions/#where-are-errors-handled). This has support for `exit_code`'s which can be appropriately bubbled through click's sys.exit (https://click.palletsprojects.com/en/7.x/commands/#command-return-values).

**Comment by jecluis** [comment]
dropping this on the floor as a major refactor is incoming.

**Pull request closed without merging**